### DISCUSSION TITLE
Dear, friend.

### DISCUSSION THREAD

**Discussion by TraderLife18** [trickerer01/Ruxx#5] [Q&A] [answered]
I am trying to download rule34hentai.net.

I replaced the module with rn.

I typed "tracer" in tags and push search.

The red light is on and the search is not working.

rx works fine.

did i do something wrong?

**Comment by trickerer01** [comment, marked as answer]
`rule34hentai.net` uses Cloudflare protection which is impossible to bypass in a convenient way currently.
There are two general approaches to download from there.

1. **Proxy**. Some proxies will automatically solve this problem for you, all you need to do is find one that bypasses Cloudflare and enter it in Ruxx's proxy window.
2. **Cookies**. You can just tell the website that you got authorized by Cloudflare once you open it in your browser You'll have to set both `user-agent` header and `cf_clearance` cookie. [In the README](https://github.com/Trickerer01/Ruxx#misc--tools) it is described in more detail.

**Reply by TraderLife18** [reply]
![1](https://user-images.githubusercontent.com/107645137/176081286-058d0603-7ab5-499b-98cc-cb99d8081758.png)

log:
Thread-2: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-3: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-4: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-5: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-6: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-7: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-8: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-9: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-10: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/

===============================================================================================
[10:30:55] Starting rn_manual

1 tasks scheduled

===============================================================================================
task 1 in progress...
tags: tracer
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 1
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 2
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 3
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 4
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 5
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 6
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 7
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 8
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 9
catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent. Reconnecting... 10
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/1
ERROR: GetItemsQueSize: unable to retreive html

Interrupted by <class 'app_defines.ThreadInterruptException'>!

Thread-12: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-13: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-14: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-15: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-16: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-17: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-18: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-19: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-20: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-21: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/tracer order=id_desc/
Thread-22: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-23: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-24: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-25: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-26: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-27: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-28: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-29: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-30: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-31: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-32: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-33: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-34: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-35: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/
Thread-36: catched <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Temporarily Unavailable for url: https://rule34hentai.net/post/list/Tracer%20order=id_desc/. Reconnecting... 1
Unable to connect. Aborting https://rule34hentai.net/post/list/Tracer order=id_desc/

The user-agent was entered as described.

cf-clealance also found what was issued to me and put it in.

Still not working

I am Korean

**Reply by TraderLife18** [reply]
Is there any program I need to install together?

**Reply by trickerer01** [reply]
> Is there any program I need to install together?

Of course not.

> Still not working

I'll recheck it on my end

**Reply by trickerer01** [reply]
Using Ruxx 1.1.195 Release. Well it downloads normally. The `check` button seems to not work properly with proxy though.
Keep in mind that these cf cookies have a rather short expiration time. In the past it was 1 hour, looks to be even shorter now.

![Ruxx195r](https://user-images.githubusercontent.com/76029665/176088483-573d6550-4ac1-4355-96a9-e6832a2d04da.JPG)

**Reply by trickerer01** [reply]
Your particular problem is most likely here:
> catched <class 'requests.exceptions.InvalidHeader'>: Invalid return character or leading space in header: User-Agent.

If you look at cmdline arguments you can see instead of slashes it displays invalid characters.
~~What locale is your OS?~~ Korean, right

**Reply by trickerer01** [reply]
Try with 1.1.197
